driver-core/src/main/com/mongodb/internal/session/BaseClientSessionImpl.java

```diff
@@ -59,6 +60,18 @@ public void setPinnedMongosAddress(@Nullable final ServerAddress address) {
         pinnedMongosAddress = address;
     }
 
+    @Override
+    public BsonDocument getRecoveryToken() {
+        return recoveryToken;
+    }
+
+    @Override
+    public void setRecoveryToken(final BsonDocument recoveryToken) {
+        if (recoveryToken != null) {
```

Comment:
With this conditional there is no way to unset the recovery token after the transaction is no longer active, which is different from mongos pinning (supports unpinning).  Let's make it the caller's responsibility to check for null if necessary.

Reply:
Done

Reply:
I don't see where this is done.  Can you elaborate?

Reply:
I reverted this in an earlier commit. I have removed the conditional in `setRecoveryToken` and moved it to the caller.